pkg/controller/petset/pet_set.go

```diff
@@ -0,0 +1,342 @@
+
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(glog.Infof)
+	eventBroadcaster.StartRecordingToSink(kubeClient.Events(""))
+	recorder := eventBroadcaster.NewRecorder(api.EventSource{Component: "petset"})
+	pc := &apiServerPetClient{kubeClient, recorder, &defaultPetHealthChecker{}}
+
+	psc := &PetSetController{
+		kubeClient:       kubeClient,
+		blockingPetStore: newUnHealthyPetTracker(),
+		newSyncer: func(blockingPet *pcb) *petSyncer {
+			return &petSyncer{pc, blockingPet}
+		},
+		queue: workqueue.New(),
+	}
+
+	// TODO: Swap this with the shared informer.
+	psc.podStore.Store, psc.podController = framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return psc.kubeClient.Pods(api.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return psc.kubeClient.Pods(api.NamespaceAll).Watch(options)
+			},
+		},
+		&api.Pod{},
+		resyncPeriod,
+		framework.ResourceEventHandlerFuncs{
+			// lookup the petset and enqueue
+			AddFunc: psc.addPod,
+			// lookup current and old petset if labels changed
+			UpdateFunc: psc.updatePod,
+			// lookup petset accounting for deletion tombstones
+			DeleteFunc: psc.deletePod,
+		})
+
+	psc.psStore.Store, psc.psController = framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return psc.kubeClient.Apps().PetSet(api.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return psc.kubeClient.Apps().PetSet(api.NamespaceAll).Watch(options)
+			},
+		},
+		&apps.PetSet{},
+		petSetResyncPeriod,
+		framework.ResourceEventHandlerFuncs{
+			AddFunc: psc.enqueuePetSet,
+			UpdateFunc: func(old, cur interface{}) {
+				oldPS := old.(*apps.PetSet)
+				curPS := cur.(*apps.PetSet)
+				if oldPS.Status.Replicas != curPS.Status.Replicas {
+					glog.V(4).Infof("Observed updated replica count for PetSet: %v, %d->%d", curPS.Name, oldPS.Status.Replicas, curPS.Status.Replicas)
+				}
+				psc.enqueuePetSet(cur)
+			},
+			DeleteFunc: psc.enqueuePetSet,
+		},
+	)
+
+	// TODO: Watch volumes
+
+	psc.podStoreSynced = psc.podController.HasSynced
+	psc.syncHandler = psc.Sync
+	return psc
+}
+
+func (psc *PetSetController) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	glog.Infof("Starting petset controller")
+	go psc.podController.Run(stopCh)
+	go psc.psController.Run(stopCh)
+	for i := 0; i < workers; i++ {
+		go wait.Until(psc.worker, time.Second, stopCh)
+	}
+	<-stopCh
+	glog.Infof("Shutting down petset controller")
+	psc.queue.ShutDown()
+}
+
+// addPod adds the petset for the pod to the sync queue
+func (psc *PetSetController) addPod(obj interface{}) {
+	pod := obj.(*api.Pod)
+	glog.Infof("Pod %s created, labels: %+v", pod.Name, pod.Labels)
```

Comment:
We do for the sake of clarity in all controllers. 
Not at infof though. This is a wip after all. 
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/replicaset/replica_set.go#L297